test/api-test.js

```diff
@@ -51,6 +51,54 @@ describe('RippleAPI', function () {
     assert.strictEqual(error.inspect(), '[RippleError(mess, { data: 1 })]');
   });
 
+  describe('pagination', function () {
+
+    describe('hasNextPage', function () {
+
+      it('returns true when there is another page', function () {
+        return this.api.request('ledger_data').then(response => {
+            assert(this.api.hasNextPage(response));
+          }
+        );
+      });
+
+      it('returns false when there are no more pages', function () {
+        return this.api.request('ledger_data').then(response => {
+          return this.api.requestNextPage('ledger_data', {}, response);
+        }).then(response => {
+          assert(!this.api.hasNextPage(response));
+        });
+      });
+
+    });
+
+    describe('requestNextPage', function () {
+
+      it('requests the next page', function () {
+        return this.api.request('ledger_data').then(response => {
+          return this.api.requestNextPage('ledger_data', {}, response);
+        }).then(response => {
+          assert.equal(response.state[0].index, '000B714B790C3C79FEE00D17C4DEB436B375466F29679447BA64F265FD63D731')
```

Comment:
Since the value comes from the fixture, I think we want to compare with this hardcoded value in order to test that the next page of data was actually retrieved.